src/styles/getTooltipStyles.js

```diff
@@ -110,6 +114,15 @@ export default function getTooltipStyles(
             ...spacing.ph2,
             zIndex: variables.tooltipzIndex,
 
+            // Generate wrapper's width using text's offsetWidth based on maxWidth prop
+            width: tooltipTextWidth < maxWidth
+
+                // Add horizontal padding to the text width to get the wrapper width.
+                // TooltipTextWidth ignores the fractions (OffsetWidth) so add 1px to fit the text properly.
+                ? tooltipTextWidth + (spacing.ph2.paddingHorizontal * 2) + 1
```

Comment:
When text is not expanding to MaxWidth which means it is taking lesser space than the MaxWidth allowed, why are we setting the width at all. let it be automatic as it was.